### PR TITLE
fix quality flaws

### PR DESCRIPTION
**Pull request by Bertk**
fix bugs, vulnerabilities

---
This change is [Reviewable](https://reviewable.io/reviews/sonaropencommunity/sonar-cxx/1153)

### CODE REVIEW

**Comment by Bertk** [comment]
@jmecosta Please check CxxCoverageSensor.java. I removed some catch blocks (still available as comment) because I was not sure whether this is still necessary and I cannot test the coverage on large source code projects.

**Comment by guwirth** [comment]
@Bertk this try/catch block are to continue after coverage errors. @jmecosta added them to switch the behavior on/off. In case the function in the try block can throw an exception we have to keep them (my opinion).

**Comment by Bertk** [comment]
@guwirth That's OK but I could not find a throw in the builder (V6.x API) but I only checked java doc.
http://javadocs.sonarsource.org/latest/apidocs/org/sonar/api/batch/sensor/coverage/NewCoverage.html

SonarQube V6.2 merges the coverage data from multiple sources which is not done with the LTS version and maybe an exception is thrown. Nevertheless we should not use a catches like this 
```
...
catch(Exception ex) {
              ...
} 
```
The java doc for the LTS does only mention exceptions for ```public static CoverageType valueOf(String name)``` but this is not used.
see http://javadocs.sonarsource.org/5.6.6/apidocs/org/sonar/api/batch/sensor/coverage/NewCoverage.html
```
Throws:
IllegalArgumentException - if this enum type has no constant with the specified name
NullPointerException - if the argument is null
```

**Comment by guwirth** [comment]
@Bertk for history see #959 and #842

**Comment by Bertk** [comment]
@guwirth Thanks for the hints. I enabled the exception handling again but used ```RuntimeException```.

File: cxx-sensors/src/main/java/org/sonar/cxx/sensors/utils/CxxReportSensor.java [hunk 1]
```diff
@@ -247,20 +278,23 @@ private void saveViolation(SensorContext sensorContext, String ruleRepoKey,
           notFoundFiles.add(normalPath);
         }
       }
-    } else { // project level
-      try {
-        NewIssue newIssue = sensorContext.newIssue().forRule(RuleKey.of(ruleRepoKey + this.language.getRepositorySuffix(), ruleId));
+    } else {
+      // project level
+//      try {
+        NewIssue newIssue = sensorContext.newIssue().forRule(
```
Comment (guwirth) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/utils/CxxReportSensor.java]: @Bertk missing try/catch

Comment (Bertk) [inline cxx-sensors/src/main/java/org/sonar/cxx/sensors/utils/CxxReportSensor.java]: I added the catch for RuntimeException.

**Comment by guwirth** [comment]
@jmecosta do you also have time to review it? Otherwise will merge it latest on Sunday evening.

**Comment by guwirth** [comment]
@Bertk thx for cleaning things up

**Pull request merged**